### ISSUE TITLE
Mod1 QC: Character limit exceeded on open text responses

### ISSUE THREAD

**Issue by KELSEYDOWLING7** [episphere/questionnaire#227]
The following Connect IDs exceeded the character length limit on open text responses (even if spaces and symbols are disregarded)

**EMPLYR1:**
1493533453
1924280178
4264097859
5072152939
5398768563
5422560778
6498891183
7528749006
7732435041
8511996300
9307929352
9730045736
4996539901

**WORK7:**
8707233264

**EMPLYR2:**
2395045862
3450934086
7743878390

**EMPLYRSRVCE2:**
4658829003
6869843940

**Comment by anthonypetersen** [comment]
@danielruss I don't think this is a Quest issue, correct? Quest isn't aware of data dictionary character limits?

**Comment by danielruss** [comment]
What question is this?  Is it possible the markdown has extra characters?

**Comment by KELSEYDOWLING7** [comment]
@danielruss 

These questions are:
EMPLYR1: 434316600, char 70

WORK7: 118061122, char 300

EMPLYR2: 988878019, char 70

EMPLYRSRVCE2: 233198706, char 300

**Comment by cusackjm** [comment]
@anthonypetersen I was referring to the character limits that are listed for each variable in the data dictionary (not the character limits of the excel DD document)

**Comment by anthonypetersen** [comment]
@cusackjm is there a different data dictionary than the excel document?

**Comment by cusackjm** [comment]
@anthonypetersen No, I thought you meant the maximum characters that the excel cell can hold, haha never mind

**Comment by danielruss** [comment]
EMPLYR1
```
[D_434316600?] What was your employer’s name for this job?
|__|size=50|

[D_118061122?] What was your longest-held job? Please be descriptive. For example, high school math teacher, emergency room nurse, automobile painter.
|__|size=50|
< -> D_279637054 >
[D_988878019?] What was your employer’s name for your longest-held job?
|__|size=50|
< -> D_530742915 >
[D_233198706?] What products were made, or services provided, by this employer? If multiple, please provide the product/services related to your work unit. For example: automotive parts, residential home construction, accounting services, pharmaceutical research.
|___|
< -> D_686172464 >

[D_857915436?] What is the address?
Line 1 (Street, PO Box, Rural Route)
|__|required id=D_284580415|
Line 2 (Apartment, Suite, Unit, Building)
|__|id=D_728926441|
City
|__|required id=D_907038282|
State
|state|id=D_970839481|
Zip
|__|id=D_379899229|
```
I don't think anything about these question limit the size of the response. (size actually sets the size of the input box, unless someone else touched this while I wasn't looking)

**Comment by danielruss** [comment]
Ran a few test to confirm.  Size effect the input box width.  It does not limit the text.

**Comment by danielruss** [comment]
so a minlen/maxlen feature?

`Enter a 5-digit zip code : |__|minlen=5 maxlen=5| `

**Comment by anthonypetersen** [comment]
yup, something like that.. unless it exists, this problem has the potential to keep happening.

**Comment by KELSEYDOWLING7** [comment]
@m-j-horner  This is the issue I was referring to

@anthonypetersen  @danielruss  Was this character limit fix implemented in all surveys or only on Module 1? We're trying to decide on a Module 4 fix for participants putting street number and name into the prompt for street number. The DD has the character limit of 6 but there are a lot of responses longer than that in BQ

**Comment by KELSEYDOWLING7** [comment]
This is definitely still an issue. People finishing Module 4 today are clearly not limited to the six characters which is causing concern for geocoding. 

![image](https://github.com/episphere/questionnaire/assets/101136647/f9c12a4b-a063-4682-8823-72e9acb4c6ed)

```
SELECT D_121490150_D_255248624, date(pt.d_264644252)
FROM `nih-nci-dceg-connect-prod-6d04.FlatConnect.module4_v1_JP`  m4
left join `nih-nci-dceg-connect-prod-6d04.FlatConnect.participants_JP`  pt
on m4.Connect_ID = pt.Connect_ID
where D_121490150_D_255248624 is not null and pt.d_663265240="231311385"
order by  date(pt.d_264644252 ) desc
```

**Comment by anthonypetersen** [comment]
I tested both scenarios I would see us wanting with the proper markdown syntax, a 6-digit max numerical field and a text field with length of 6 as a maximum, and Quest properly stopped me from continuing.
